File: app/src/main/java/captain/wonjong/fontchecker/ColorDialog.java
package captain.wonjong.fontchecker;

import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.LifecycleOwner;
import captain.wonjong.fontchecker.databinding.LayoutColorDialogBinding;
import captain.wonjong.fontchecker.model.ColorModel;

public class ColorDialog {
    private Context                   mContext;
    private Dialog                    mDialog;

    private ColorModel                mModel;
    private LayoutColorDialogBinding  mBinding;

    // Singleton
    private static ColorDialog uniqueInstance = new ColorDialog();
    public static ColorDialog getInstance() {
        return uniqueInstance;
    }

    public ColorDialog show(@NonNull Context context) {
        mContext = context;

        mDialog = new Dialog(mContext, R.style.ChangeDialogStyle);
        mDialog.setCancelable(true);
        mDialog.setCanceledOnTouchOutside(true);

        mModel = new ColorModel();
        mBinding = DataBindingUtil.inflate(LayoutInflater.from(mDialog.getContext()), R.layout.layout_color_dialog, null, false);
        mDialog.setContentView(mBinding.getRoot());

        mBinding.setLifecycleOwner((LifecycleOwner) mContext);
        mBinding.setModel(mModel);


        try {
            mDialog.show();
        } catch (Exception e) {
            mDialog = null;
        }

        return getInstance();
    }

    public ColorDialog onConfirm(View.OnClickListener onClickListener) {
        mBinding.btnConfirm.setOnClickListener(onClickListener);
        return getInstance();
    }

    public ColorDialog onCancel(View.OnClickListener onClickListener) {
        mBinding.btnCancel.setOnClickListener(onClickListener);
        return getInstance();
    }

    public int getColor() {
        return mModel.color.getValue();
    }

    public int getRed() {
        return mModel.red.getValue();
    }

    public int getGreen() {
        return mModel.green.getValue();
    }

    public int getBlue() {
        return mModel.blue.getValue();
    }

    public void dismiss() {
        if (mDialog != null) {
            mDialog.dismiss();
        }
    }
}

File: app/src/main/java/com/yujin/fontchecker/ColorDialog.kt
package com.yujin.fontchecker

import android.app.Dialog
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import androidx.databinding.DataBindingUtil
import androidx.lifecycle.LifecycleOwner
import com.yujin.fontchecker.databinding.LayoutColorDialogBinding
import com.yujin.fontchecker.model.ColorModel

class ColorDialog(private val context: Context) {
    private var dialog: Dialog? = null
    private var colorModel: ColorModel? = null
    private var binding: LayoutColorDialogBinding? = null

    val color: Int?
        get() = colorModel?.color?.value
    val red: Int?
        get() = colorModel?.red?.value
    val green: Int?
        get() = colorModel?.green?.value
    val blue: Int?
        get() = colorModel?.blue?.value

    fun show() {
        colorModel = ColorModel()
        dialog = Dialog(context, R.style.ChangeDialogStyle).apply {
            setCanceledOnTouchOutside(true)
        }

        binding = DataBindingUtil.inflate(LayoutInflater.from(dialog?.context), R.layout.layout_color_dialog, null, false)
        binding?.run {
            setLifecycleOwner(context as LifecycleOwner)
            model = colorModel
        }

        dialog?.setContentView(binding!!.root)

        try {
            dialog?.show()
        } catch (e: Exception) {
            dialog = null
        }
    }

    fun onConfirm(onClickListener: (View) -> Unit) {
        binding?.btnConfirm?.setOnClickListener(onClickListener)
    }

    fun onCancel(onClickListener: (View) -> Unit) {
        binding?.btnCancel?.setOnClickListener(onClickListener)
    }

    fun dismiss() {
        dialog?.dismiss()
    }
}